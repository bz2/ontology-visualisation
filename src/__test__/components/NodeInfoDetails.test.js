import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import NodeInfoDetails from '../../components/NodeInfoDetails'

const setup = () => {
  const props = {
    availableNodesNormalised: {
      abc: {
        id: 'abc',
        label: 'label',
        rdfAbout: 'rdfAbout',
        skosComment: 'skosComment',
        skosDefinition: 'skosDefinition',
        skosExample: 'skosExample'
      },
      cde: {
        id: 'cde',
        label: 'label2',
        rdfAbout: 'rdfAbout',
        skosComment: 'skosComment',
        skosDefinition: 'skosDefinition',
        skosExample: 'skosExample'
      }
    },
    nodesConnections: [{
      from: 'abc',
      fromLabel: 'label',
      to: 'cde',
      toLabel: 'label2',
      label: 'is child of',
      edgeId: 'abc'
    }],
    nodeId: 'abc'
  }

  const component = shallow(<NodeInfoDetails {...props} />)

  return {
    component,
    props
  }
}

describe('NodeInfoDetails', () => {
  it('should match snapshot ', () => {
    const {
      component
    } = setup()

    expect(toJson(component)).toMatchSnapshot()
  })
})
