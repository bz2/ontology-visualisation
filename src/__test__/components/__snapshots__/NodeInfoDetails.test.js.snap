// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NodeInfoDetails should match snapshot  1`] = `
<Connect
  actions={
    Object {
      "addToArray": [Function],
      "removeFromArray": [Function],
      "removeFromObject": [Function],
      "setStoreState": [Function],
    }
  }
  availableNodesNormalised={
    Object {
      "abc": Object {
        "id": "abc",
        "label": "label",
        "rdfAbout": "rdfAbout",
        "skosComment": "skosComment",
        "skosDefinition": "skosDefinition",
        "skosExample": "skosExample",
      },
      "cde": Object {
        "id": "cde",
        "label": "label2",
        "rdfAbout": "rdfAbout",
        "skosComment": "skosComment",
        "skosDefinition": "skosDefinition",
        "skosExample": "skosExample",
      },
    }
  }
  mapToProps={[Function]}
  nodeId="abc"
  nodesConnections={
    Array [
      Object {
        "edgeId": "abc",
        "from": "abc",
        "fromLabel": "label",
        "label": "is child of",
        "to": "cde",
        "toLabel": "label2",
      },
    ]
  }
>
  <Component />
</Connect>
`;
